web_src/js/components/ViewFileTree.vue

```diff
@@ -12,27 +20,244 @@ const props = defineProps({
 });
 
 const store = createViewFileTreeStore(props);
+
+const filteredFiles = computed(() => {
+  if (!searchQuery.value) return [];
+  return filterRepoFilesWeighted(allFiles.value, searchQuery.value);
+});
+
+const treeLink = computed(() => `${props.repoLink}/src/${props.currentRefNameSubURL}`);
```

Comment:
```suggestion
const treeLink = computed(() => `${props.repoLink}/src/${encodeURIComponent(props.currentRefNameSubURL)}`);
```

Good practice in case thise string contains url-unsafe characters.

Reply:
Not encodeURIComponent, it needs path segment escaping

Reply:
Right, we have the `pathEscapeSegments` function for that.